src/Microsoft.ML.Data/Scorers/FeatureContributionCalculationTransform.cs

```diff
@@ -15,39 +15,26 @@
 using Microsoft.ML.Runtime.Model;
 using Microsoft.ML.Runtime.Numeric;
 
-[assembly: LoadableClass(typeof(IDataScorerTransform), typeof(FeatureContributionCalculationTransform), typeof(FeatureContributionCalculationTransform.Arguments),
-    typeof(SignatureDataScorer), "Feature Contribution Transform", "fct", "FeatureContributionCalculationTransform", MetadataUtils.Const.ScoreColumnKind.FeatureContribution)]
+[assembly: LoadableClass(typeof(IDataScorerTransform), typeof(FeatureContributionScorer), typeof(FeatureContributionScorer.Arguments),
+    typeof(SignatureDataScorer), "Feature Contribution Scorer", "fcc", "FeatureContributionCalculationScorer", MetadataUtils.Const.ScoreColumnKind.FeatureContribution)]
 
-[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(FeatureContributionCalculationTransform), typeof(FeatureContributionCalculationTransform.Arguments),
-    typeof(SignatureBindableMapper), "Feature Contribution Mapper", "fct", MetadataUtils.Const.ScoreColumnKind.FeatureContribution)]
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(FeatureContributionScorer), typeof(FeatureContributionScorer.Arguments),
+    typeof(SignatureBindableMapper), "Feature Contribution Mapper", "fcc", MetadataUtils.Const.ScoreColumnKind.FeatureContribution)]
 
-[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(FeatureContributionCalculationTransform), null, typeof(SignatureLoadModel),
-    "Feature Contribution Mapper", FeatureContributionCalculationTransform.MapperLoaderSignature)]
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(FeatureContributionScorer), null, typeof(SignatureLoadModel),
+    "Feature Contribution Mapper", FeatureContributionScorer.MapperLoaderSignature)]
 
 namespace Microsoft.ML.Runtime.Data
 {
     /// <summary>
-    /// Feature Contribution Calculation Transform.
+    /// Used only by the command line API for scoring and calculation of feature contribution.
     /// </summary>
-    /// <remarks>
-    /// The Feature Contribution Calculation Transform scores the model on an input dataset and
-    /// computes model-specific contribution scores for each feature. See the sample below for
-    /// an example of how to compute feature importance using the Feature Contribution Calculation Transform.
-    /// </remarks>
-    /// <example>
-    /// <format type="text/markdown">
-    /// <![CDATA[
-    /// [!code-csharp[FCT](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/FeatureContributionCalculationTransform.cs)]
-    /// ]]>
-    /// </format>
-    /// </example>
-    public sealed class FeatureContributionCalculationTransform
+    internal sealed class FeatureContributionScorer
     {
         // Apparently, loader signature is limited in length to 24 characters.
         internal const string MapperLoaderSignature = "WTFBindable";
-        private const int MaxTopBottom = 1000;
```

Comment:
>MaxTopBottom [](start = 26, length = 12)

why removing this?

Reply:
Rogan asked that we don't fix a boundary for bottom and top parameters.

---
In reply to: [243094076](https://github.com/dotnet/machinelearning/pull/1847#discussion_r243094076) [](ancestors = 243094076)